feat: improve ziggle ux

tkjftkjf opened the pull request:
- 공지 상세 페이지에 SafeArea 적용
- 안드로이드에서 공지 상세 페이지 빠져나올 때 잔상 남는 문제 해결
  - 안드로이드에서도 IOS 스타일 애니메이션 적용
- 새로고침 바 색상 보라 -> 주황 변경 

Closes #603 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## 릴리스 노트

* **New Features**
  * 안전 영역(Safe Area) 지원으로 디바이스 노치 및 하단 영역 호환성 개선
  * 색상 테마 개선으로 디자인 시스템 일관성 강화

* **Style**
  * UI 요소 포매팅 및 레이아웃 조정

* **Chores**
  * 내부 의존성 관리 최적화

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tkjftkjf committed:
🚸 feat: add safeArea at notice detail page

tkjftkjf committed:
🚸 feat: set colors of refresh indicator

- 기존의 보라색 인디케이터를 Palette.primary 색상으로 변경하였습니다.

tkjftkjf committed:
fix: remove afterimage of notice detail page by forcing to use IOS style animation

- change import to material to cupertino
- change cotice detail page router from AutoRoute to CustomRoute
- add new custom route builder for IOS style animation

dawnfire05 reviewed (commented):
Refresh Indicator 색 app.dart에 있는 theme으로 바꿀 순 없어요?
Refresh Indicator마다 색 바꿔주는거 좀 그런데... 빼먹을 수도 있고

dawnfire05 commented:
@tkjftkjf 그러고보니, 통일성을 위해 화면 넘어가는 애니메이션 다른 화면들에도 적용해야 하지 않을까요?

dawnfire05 commented:
빌드 에러 해결된거에요?

tkjftkjf commented:
아니요 로컬에서 생성된 파일 문젠가 해서 rm -rf하고 다시 클론 했는데 리베이스 사라져서 다시 한거에요.

그리고 페이지 애니메이션 관련해서 제 기억으론 같은 애니메이션인 화면이 설정 페이지인데, 그 화면은 개인적으론 IOS 스타일이면 조금 어색할 것 같아요

tkjftkjf committed:
feat: implement ZiggleRefreshIndicator for consistent refresh behavior across pages

- 다양한 페이지에서 사용되는 refresh indicator의 테마 일관성을 위해 ZiggleRefreshWidget을 구현하고 기존의 refresh indicator를 전부 ziggle refresh indicator로 교체함

2paperstar commented:
머티리얼 페이지 트랜지션을 수정해서 해결할 수는 없는건가요..? WebView가 투명도를 지원 안 하나..

2paperstar commented:
> Refresh Indicator 색 app.dart에 있는 theme으로 바꿀 순 없어요? Refresh Indicator마다 색 바꿔주는거 좀 그런데... 빼먹을 수도 있고

이거 AppTheme인가에 RefreshIndicatorTheme이 있을걸요?

tkjftkjf commented:
ThemeData에 RefreshIndicatorTheme이 없는데, 있는 IndicatorTheme이랑 ProgressIndicatorTheme으로도 해결이 안되어서 커스텀 위젯으로 만들게 되었습니다.

2paperstar commented:
> ThemeData에 RefreshIndicatorTheme이 없는데, 있는 IndicatorTheme이랑 ProgressIndicatorTheme으로도 해결이 안되어서 커스텀 위젯으로 만들게 되었습니다.

왜지..? 안 될 수가 없는데
수정하고 reload했는데도 적용이 안 됐었나요?

tkjftkjf commented:
> 
> 왜지..? 안 될 수가 없는데 수정하고 reload했는데도 적용이 안 됐었나요?

hot reload 하면서 해봤는데, background color는 변경이 되는데, color 자체는 ThemeData로 변경이 안되더라구요...
그래도 혹시 모르니깐 다시 한번 테스트 해볼게요.

dawnfire05 commented:
colorScheme: ColorScheme.fromSeed(seedColor: Palette.primary),
 이거로 수정

tkjftkjf committed:
refactor: provides color consistency using colorScheme instead of a custom refresh widget

tkjftkjf committed:
fix: remove custome route
